File: backend/src/main/java/com/jorge/gymprogress/model/WorkoutSet.java
package com.jorge.gymprogress.model;

import jakarta.persistence.*;

@Entity
@Table(name = "workout_sets")
public class WorkoutSet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Entrenamiento al que pertenece esta serie
    @ManyToOne(optional = false)
    @JoinColumn(name = "workout_session_id", nullable = false)
    private WorkoutSession workoutSession;

    // Ejercicio realizado en esta serie
    @ManyToOne(optional = false)
    @JoinColumn(name = "exercise_id", nullable = false)
    private Exercise exercise;

    @Column(nullable = false)
    private Integer setNumber;

    @Column(nullable = false)
    private Double weightKg;

    @Column(nullable = false)
    private Integer repetitions;

    private String notes;

    public WorkoutSet() {
    }

    public Double calcularVolumen() {
        return weightKg * repetitions;
    }

    public Long getId() {
        return id;
    }

    public WorkoutSession getWorkoutSession() {
        return workoutSession;
    }

    public Exercise getExercise() {
        return exercise;
    }

    public Integer getSetNumber() {
        return setNumber;
    }

    public Double getWeightKg() {
        return weightKg;
    }

    public Integer getRepetitions() {
        return repetitions;
    }

    public String getNotes() {
        return notes;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setWorkoutSession(WorkoutSession workoutSession) {
        this.workoutSession = workoutSession;
    }

    public void setExercise(Exercise exercise) {
        this.exercise = exercise;
    }

    public void setSetNumber(Integer setNumber) {
        this.setNumber = setNumber;
    }

    public void setWeightKg(Double weightKg) {
        this.weightKg = weightKg;
    }

    public void setRepetitions(Integer repetitions) {
        this.repetitions = repetitions;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }
}

File: frontend/lib/models/workout_set.dart
class WorkoutSet {
  final int id;
  final int workoutSessionId;
  final int exerciseId;
  final String exerciseName;
  final String muscleGroup;
  final int setNumber;
  final double weightKg;
  final int repetitions;
  final double volume;
  final String notes;

  WorkoutSet({
    required this.id,
    required this.workoutSessionId,
    required this.exerciseId,
    required this.exerciseName,
    required this.muscleGroup,
    required this.setNumber,
    required this.weightKg,
    required this.repetitions,
    required this.volume,
    required this.notes,
  });

  factory WorkoutSet.fromJson(Map<String, dynamic> json) {
    return WorkoutSet(
      id: json['id'],
      workoutSessionId: json['workoutSessionId'],
      exerciseId: json['exerciseId'],
      exerciseName: json['exerciseName'] ?? '',
      muscleGroup: json['muscleGroup'] ?? '',
      setNumber: json['setNumber'] ?? 0,
      weightKg: (json['weightKg'] ?? 0).toDouble(),
      repetitions: json['repetitions'] ?? 0,
      volume: (json['volume'] ?? 0).toDouble(),
      notes: json['notes'] ?? '',
    );
  }
}